File: justmeetbackend/src/main/java/it/unicam/justmeetbackend/databasefake/Evento.java
package it.unicam.justmeetbackend.databasefake;

import org.joda.time.DateTime;

public class Evento {
    
    private String id;
    private String titolo;
    private String desc;
    private int partecipanti;
    private String idTopic;
    private String idCreator;
    private String nomeComune;
    private boolean isApproved;
    private DateTime inizioEvento;
    private DateTime fineEvento;
   

    public Evento() {
    };

    public Evento(final String id, final String t, final String d, final int partecipanti, final String topic,
            final String creatore, final String luogo, String inizioEvento, String fineEvento, boolean isApp) {
        this.id = id;
        this.titolo = t;
        this.desc = d;
        this.partecipanti = partecipanti;
        this.idTopic = topic;
        this.idCreator = creatore;
        this.inizioEvento = new DateTime(inizioEvento);
        this.fineEvento = new DateTime(fineEvento);
        this.nomeComune = luogo;
        this.isApproved = isApp;
    }

    public String getId() {
        return id;
    }

    public void setId(final String id) {
        this.id = id;
    }

    public String getTitolo() {
        return titolo;
    }

    public void setTitolo(final String titolo) {
        this.titolo = titolo;
    }

    public String getDesc() {
        return desc;
    }

    public void setDesc(final String desc) {
        this.desc = desc;
    }

    public String getUser() {
        return idCreator;
    }

    public void setUser(final String user) {
        this.idCreator = user;
    }

    public int getPartecipanti() {
        return partecipanti;
    }

    public void setPartecipanti(final int partecipanti) {
        this.partecipanti = partecipanti;
    }

    public String getIdTopic() {
        return idTopic;
    }

    public void setIdTopic(final String idTopic) {
        this.idTopic = idTopic;
    }

    public String getIdCreator() {
        return idCreator;
    }

    public void setIdCreator(final String idCreator) {
        this.idCreator = idCreator;
    }

    /**
     * @return the inizioEvento
     */
    public String getInizioEvento() {
        return inizioEvento.toString();
    }

    /**
     * @param inizioEvento the inizioEvento to set
     */
    public void setInizioEvento(String inizioEvento) {
        this.inizioEvento = new DateTime(inizioEvento);
    }

    /**
     * @return the fineEvento
     */
    public String getFineEvento() {
        return fineEvento.toString();
    }

    /**
     * @param fineEvento the fineEvento to set
     */
    public void setFineEvento(String fineEvento) {
        this.fineEvento = new DateTime(fineEvento);
    }
    public boolean isApproved() {
        return isApproved;
    }

    public void setApproved(boolean isApproved) {
        this.isApproved = isApproved;
    }

    public String getNomeComune() {
        return nomeComune;
    }

    public void setNomeComune(String nomeComune) {
        this.nomeComune = nomeComune;
    }
     
}

File: justmeetfrontend/lib/classi/evento.dart
class Evento {
  
      String id;
      String titolo;
      String descrizione;
      int partecipanti; 
      String idTopic;   
      String idCreatore; 
      String idComune;
      bool isApproved;
      DateTime inizioEvento;
      DateTime fineEvento;
      List<String> iscrizioni;
      

      
    Evento(String titolo, String descrizione, int partecipanti, String idTopic, String idCreatore, String idComune, String inizioEvento ,String fineEvento)
    {
      this.titolo = titolo;
      this.descrizione = descrizione;
      this.partecipanti = partecipanti;
      this.idTopic = idTopic;
      this.idCreatore = idCreatore;
      this.idComune = idComune;
      this.inizioEvento = DateTime.parse(inizioEvento);
      this.fineEvento = DateTime.parse(fineEvento);
    }

  Map<String, dynamic> toMap() 
  {
    var map = new Map<String, dynamic>();
    map["titolo"] = this.titolo;
    map["descrizione"] = this.descrizione;
    map["partecipanti"] = this.partecipanti;
    map["idTopic"] = this.idTopic;
    map["idCreatore"] = this.idCreatore;
    map["idComune"] = this.idComune;
    map["inizioEvento"] = this.inizioEvento.toIso8601String();
    map["fineEvento"] = this.fineEvento.toIso8601String();
    return map;
  }


Evento.fromjson(Map<String, dynamic> json)
        :
          id = json['id'],
          titolo = json['titolo'],
          descrizione = json['descrizione'],
          partecipanti = json['partecipanti'],
          idTopic = json['idTopic'],
          idCreatore = json['idCreatore'],
          idComune = json['idComune'],
          isApproved = json['isApproved'],
          inizioEvento = DateTime.parse(json['inizioEvento']),
          fineEvento = DateTime.parse(json['fineEvento']);
          //iscrizioni = 
            
  
	String getId() {
		return this.id;
	}

	void setId(String id) {
		this.id = id;
	}

	String getNome() {
		return this.titolo;
	}

	void setNome(String titolo) {
		this.titolo = titolo;
	}

	String getDesc() {
		return this.descrizione;
	}

	void setDesc(String desc) {
		this.descrizione = desc;
	}

	int getMaxPartecipanti() {
		return this.partecipanti;
	}

	void setMaxPartecipanti(int maxPartecipanti) {
		this.partecipanti = maxPartecipanti;
	}

@override
  String toString() {
    String infoEvento;
    infoEvento = "ID: "+ id +"\n"+
                  "Titolo: "+ titolo + "\n"+
                  "Descrizione: "+ descrizione + "\n"+
                  "Partecipanti: "+ partecipanti.toString() + "\n"+
                  "Topic: "+ idTopic + "\n"+
                  "Comune: "+ idComune + "\n"+
                  "Creatore: "+ idCreatore + "\n"+
                  "Inizio:"+ inizioEvento.toIso8601String() + "\n"+
                  "Fine: "+ fineEvento.toIso8601String() + "\n"+
                  "Approvato: " + isApproved.toString();
    return infoEvento;
  }
}

      